app/test/testMetadata.js

```diff
@@ -0,0 +1,161 @@
+const chai = require('chai');
+const chaiHttp = require('chai-http');
+const fetchMock = require('fetch-mock');
+const {app} = require('../service');
+
+chai.use(chaiHttp);
+chai.should();
+
+const exampleMetadata = {
+  description: 'An example description',
+  favicon_url: 'http://www.example.com/rich-icon.png',
+  icon_url: 'http://www.example.com/rich-icon.png',
+  image_url: 'http://www.example.com/preview.png',
+  images: [
+    {
+      'url': 'http://www.example.com/preview.png',
+      'entropy': 1,
+      'height': 500,
+      'width': 500,
+    }
+  ],
+  original_url: 'http://www.example.com/good',
+  provider_url: 'http://www.example.com/good',
+  title: 'An Example Page',
+  url: 'http://www.example.com/good'
+};
+
+const goodExampleUrl = 'http://www.example.com/good';
+const badExampleUrl = 'http://www.example.com/bad';
+
+
+describe('Metadata API Tests', function() {
+  before(function() {
+    fetchMock.mock(
+      goodExampleUrl,
+      `
+        <html>
+          <head>
+            <title>${exampleMetadata.title}</title>
+            <meta name="description" content="${exampleMetadata.description}" />
+            <meta name="thumbnail" content="${exampleMetadata.images[0].url}" />
+            <link rel="icon" href="http://www.example.com/rich-icon.png" />
+          </head>
+          <body></body>
+        </html>
+      `
+    );
+    fetchMock.mock(badExampleUrl, 500);
+  });
+
+  after(function() {
+    fetchMock.restore();
+  });
+
+  it('should raise 400 if missing JSON body', (done) => {
+    chai.request(app)
+      .post('/')
+      .end((err, res) => {
+        res.should.have.status(400);
+        done();
+      });
+  });
+
+  it('should raise 400 if missing urls parameter', (done) => {
+    chai.request(app)
+      .post('/')
+      .send(JSON.stringify({}))
+      .end((err, res) => {
+        res.should.have.status(400);
+        done();
+      });
+  });
```

Comment:
I'm marginally indifferent, but you can also convert these to Promises (instead of callback model) by adding a `.then()` and optional `.catch()` (plus returning the `chai.request(app)` promise chain:

```js
  it('should raise 400 if missing urls parameter', () => {
    return chai.request(app)
      .post('/')
      .send(JSON.stringify({}))
      .then((res) => res)
      .catch((err) => {
        err.should.have.status(400);
      });
  });
```

Either way [seemingly] works, just depends on if you fancy Promises over callbacks.